statsmodels/tsa/statespace/mlemodel.py

```diff
@@ -0,0 +1,797 @@
+    """
+    Class to hold results from fitting a state space model.
+
+    Parameters
+    ----------
+    model : Model instance
+        The fitted model instance
+
+    Attributes
+    ----------
+    aic : float
+        Akaike Information Criterion
+    bic : float
+        Bayes Information Criterion
+    bse : array
+        The standard errors of the parameters. Computed using the numerical
+        Hessian.
+    cov_params : array
+        The variance / covariance matrix. Computed using the numerical Hessian.
+    hqic : array
+        Hannan-Quinn Information Criterion
+    llf : float
+        The value of the log-likelihood function evaluated at `params`.
+    model : Model instance
+        A reference to the model that was fit.
+    nobs : float
+        The number of observations used to fit the model.
+    params : array
+        The parameters of the model.
+    pvalues : array
+        The p-values associated with the z-statistics of the coefficients.
+        Note that the coefficients are assumed to have a Normal distribution.
+    scale : float
+        This is currently set to 1.0 and not used by the model or its results.
+    sigma2 : float
+        The variance of the residuals.
+    zvalues : array
+        The z-statistics for the coefficients.
+
+    Methods
+    -------
+    conf_int
+    f_test
+    fittedvalues
+    forecast
+    load
+    predict
+    remove_data
+    resid
+    save
+    summary
+    t_test
+    wald_test
+    """
+    def __init__(self, model, *args, **kwargs):
+        self.data = model.data
+
+        # Save the model output
+        self._endog_names = model.endog_names
+        self._exog_names = model.endog_names
+        self._params = model.params
+        self._params_names = model.params_names
+        self._model_names = model.model_names
+        self._model_latex_names = model.model_latex_names
+
+        # Associate the names with the true parameters
+        params = pd.Series(self._params, index=self._params_names)
+
+        # Initialize the Statsmodels model base
+        tsbase.TimeSeriesModelResults.__init__(self, model, params,
+                                               normalized_cov_params=None,
+                                               scale=1., *args, **kwargs)
+
+        # Initialize the statespace representation
+        super(MLEResults, self).__init__(model)
+
+        # Setup the cache
+        self._cache = resettable_cache()
+
+    @cache_readonly
+    def aic(self):
+        # return -2*self.llf + 2*self.params.shape[0]
+        return aic(self.llf, self.nobs, self.params.shape[0])
+
+    @cache_readonly
+    def bic(self):
+        # return -2*self.llf + self.params.shape[0]*np.log(self.nobs)
+        return bic(self.llf, self.nobs, self.params.shape[0])
+
+    @cache_readonly
+    def bse(self):
+        return np.sqrt(np.diagonal(self.cov_params))
+
+    @cache_readonly
+    def cov_params(self):
```

Comment:
in `statsmodels.base` model (LikelihoodModel) `cov_params` is a method not a cached attribute.
I think it should work to rename this to `cov_params_default`. At least it would work for the base super classes, but I don't know the intermediate classes in the class hierarchy.